pyvrp/SolveParams.py

```diff
@@ -0,0 +1,96 @@
+import pathlib
+from typing import Iterable, Type, Union
+
+import tomli
+
+import pyvrp.search
+from pyvrp.GeneticAlgorithm import GeneticAlgorithmParams
+from pyvrp.PenaltyManager import PenaltyParams
+from pyvrp.Population import PopulationParams
+from pyvrp.search import (
+    NODE_OPERATORS,
+    ROUTE_OPERATORS,
+    NeighbourhoodParams,
+    NodeOperator,
+    RouteOperator,
+)
+
+
+class SolveParams:
+    """
+    Solver parameters for PyVRP's hybrid genetic search algorithm.
+
+    Parameters
+    ----------
+    genetic
+        Genetic algorithm parameters.
+    penalty
+        Penalty parameters.
+    population
+        Population parameters.
+    neighbourhood
+        Neighbourhood parameters.
+    node_ops
+        Node operators to use in the search.
+    route_ops
+        Route operators to use in the search.
+    """
+
+    def __init__(
+        self,
+        genetic: GeneticAlgorithmParams = GeneticAlgorithmParams(),
+        penalty: PenaltyParams = PenaltyParams(),
+        population: PopulationParams = PopulationParams(),
+        neighbourhood: NeighbourhoodParams = NeighbourhoodParams(),
+        node_ops: Iterable[Type[NodeOperator]] = NODE_OPERATORS,
+        route_ops: Iterable[Type[RouteOperator]] = ROUTE_OPERATORS,
+    ):
+        self._genetic = genetic
+        self._penalty = penalty
+        self._population = population
+        self._neighbourhood = neighbourhood
+        self._node_ops = node_ops
+        self._route_ops = route_ops
+
+    @property
+    def genetic(self):
+        return self._genetic
+
+    @property
+    def penalty(self):
+        return self._penalty
+
+    @property
+    def population(self):
+        return self._population
+
+    @property
+    def neighbourhood(self):
+        return self._neighbourhood
+
+    @property
+    def node_ops(self):
+        return self._node_ops
+
+    @property
+    def route_ops(self):
+        return self._route_ops
+
+    @classmethod
+    def from_file(cls, loc: Union[str, pathlib.Path]):
+        """
+        Loads the solver parameters from a TOML file.
+        """
+        with open(loc, "rb") as fh:
+            data = tomli.load(fh)
+
+        gen_params = GeneticAlgorithmParams(**data.get("genetic", {}))
+        pen_params = PenaltyParams(**data.get("penalty", {}))
+        pop_params = PopulationParams(**data.get("population", {}))
+        nb_params = NeighbourhoodParams(**data.get("neighbourhood", {}))
+        node_ops = [getattr(pyvrp.search, op) for op in data["node_ops"]]
+        route_ops = [getattr(pyvrp.search, op) for op in data["route_ops"]]
```

Comment:
If not passed, should we then not default to `NODE_OPERATORS`, or `ROUTE_OPERATORS`? That'd match the constructor.